enhancements/kubelet/user-namespaces-support.md

```diff
@@ -336,8 +336,44 @@ Thus, we need a method for the kube-apiserver to know for certain a node joining
 A way to do this that also could set a precedent for Openshift to have the apiserver be more conscious about kubelet version would be an openshift-apiserver
 extension that refuses to let a kubelet join a cluster or get leases if it is not new enough.
 
-For such a feature, there could be an extension to the NodeConfig object called `minimumKubeletVersion`. This field would be set on the cluster level. Practically, the
-openshift-apiserver could then query the `Node.Status.NodeSystemInfo.KubeletVersion` and verify it is above the `minimumKubeletVersion`.
+For such a feature, there could be an extension to the NodeConfig object called `minimumKubeletVersion`. This field would be set on the cluster level.
+
+1: I actually can't find any reference to this, but this is the way it *should* work, and thus we will rely on this.
+
+##### MinimumKubeletVersion API
+
+The API will be added to the NodeSpec object, which is a singleton cluster-wide:
+
+```
+type NodeSpec struct {
+	...
+	// MinimumKubeletVersion is the lowest version of a kubelet that can join the cluster
+	// +kubebuilder:validation:Pattern=`^([0-9]*\.[0-9]*\.[0-9]*$`
+	// +optional
+	MinimumKubeletVersion string `json:"minimumKubeletVersion,omitempty"`
+}
+```
+
+##### MinimumKubeletVerison authorization plugin
+
+The operative piece of this feature will be an authorization plugin patched into the kube-apiserver. It will be run against requests coming from
+kubelets, and if kubelet version reported through `Node.Status.NodeSystemInfo.KubeletVersion` is lower than `MinimumKubeletVersion` then the kube-apiserver
+will deny the kubelet from gaining access to any resources that aren't Node get/update and `subjectaccessreviews`. In other words, the kubelet can read it's node
+object and learn about what it has access to, but it's not allowed to gain access to any other API objects.
+
+This is a fairly harsh punishment for a node, as it means it's effectively lost and the cluster admin needs to manually intervene to remove it. There will also be
+some pieces to mitigate this pain for cluster admins.
+
+##### MinimumKubeletVersion admission plugin
+
+To protect nodes that are too old from being removed immediately, validation will be run on the MinimumKubeletVersion on admission to see if there are kubelets in the cluster
+that are running with a version lower than the configured version. If so, the creation will be rejected, and the admin notified that they must upgrade their nodes before applying it.
+
+##### MinimumKubeletVersion MCO awareness
+
+MCO will read the MinimumKubeletVersion and mark machines as degraded if the node is not at least MinimumKubeletVersion.
```

Comment:
> mark machines as degraded

This is fuzzy to me in how the MCO knows the kubelet version on each machine/node, how it is marked as degraded, and what the implications are of a machine being degraded.